tests/Unit/ValueObject/SearchUrlBuilderTest.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Answear\LuigisBoxBundle\Tests\Unit\ValueObject;
+
+use Answear\LuigisBoxBundle\ValueObject\Search\Context;
+use Answear\LuigisBoxBundle\ValueObject\SearchUrlBuilder;
+use PHPUnit\Framework\TestCase;
+
+class SearchUrlBuilderTest extends TestCase
+{
+    /**
+     * @test
+     */
+    public function buildValidUrlTest(): void
```

Comment:
Maybe the dataProvider will be better for testing this scenarios.

Reply:
I don't think so. Now you can see step by step which fields are being added and how the SearchUrlBuilder are being built.